bundles/org.openhab.binding.yioremote/src/main/java/org/openhab/binding/yioremote/internal/YIOremoteDockHandler.java

```diff
@@ -0,0 +1,384 @@
+                updateStatus(ThingStatus.ONLINE);
+                startwebsocketpollingthread();
+                if (authenticationjob != null) {
+                    authenticationjob.cancel(true);
+                }
+            } else {
+                yioremotedockactualstatus = YioRemoteDockHandleStatus.AUTHENTICATION_FAILED;
+            }
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Connection lost no ping from YIO DOCK");
+            updateState(GROUP_OUTPUT, STATUS_STRING_CHANNEL, UnDefType.UNDEF);
+        }
+    }
+
+    private void startwebsocketpollingthread() {
+        Runnable websocketpollingthread = new Runnable() {
+            @Override
+            public void run() {
+                if (yioremotedockactualstatus.equals(YioRemoteDockHandleStatus.AUTHENTICATION_COMPLETE)) {
+                    if (getbooleanheartbeat()) {
+                        updateChannelString(GROUP_OUTPUT, STATUS_STRING_CHANNEL,
+                                ircodereceivedhandler.getcode() + ircodereceivedhandler.getformat());
+                        logger.debug("heartbeat ok");
+                        sendMessage(YioRemoteMessages.HEARTBEAT_MESSAGE, "");
+                    } else {
+                        yioremotedockactualstatus = YioRemoteDockHandleStatus.CONNECTION_FAILED;
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                "Connection lost no ping from YIO DOCK");
+                        updateState(GROUP_OUTPUT, STATUS_STRING_CHANNEL, UnDefType.UNDEF);
+                        if (websocketpollingjob != null) {
+                            websocketpollingjob.cancel(true);
+                        }
+                    }
+                } else {
+                    if (websocketpollingjob != null) {
+                        websocketpollingjob.cancel(true);
+                    }
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "Connection lost no ping from YIO DOCK");
+                    updateState(GROUP_OUTPUT, STATUS_STRING_CHANNEL, UnDefType.UNDEF);
+                }
+            }
+        };
+        websocketpollingjob = scheduler.scheduleWithFixedDelay(websocketpollingthread, 0, 30, TimeUnit.SECONDS);
+    }
+
+    public boolean getbooleanauthenticationok() {
+        return authenticationok;
+    }
+
+    public boolean getbooleanheartbeat() {
+        boolean result = heartbeat;
+        heartbeat = false;
+        return result;
+    }
+
+    public YioRemoteDockHandleStatus getyioremotedockactualstatus() {
+        return yioremotedockactualstatus;
+    }
+
+    public void sendMessage(YioRemoteMessages messagetype, String messagepyload) {
+        if (messagetype.equals(YioRemoteMessages.AUTHENTICATE_MESSAGE)) {
```

Comment:
Better use a switch/case to check the enum fields. This way you get a compiler warning if you miss to handle any enum constant.

Reply:
done